src/storage/ExitPointsStorage.ts

```diff
@@ -0,0 +1,57 @@
+import { BlockHeight, ForkId } from "@/types/types";
+
+export class ExitPointsStorage {
+    private map: Map<ForkId, Set<BlockHeight>>;
+
+    constructor() {
+        this.map = new Map();
+    }
+
+    // ====================================
+    // CREATE
+    // ====================================
+
+    storeExitPoint(forkId: ForkId, blockHeight: BlockHeight): Set<BlockHeight> {
+        const exitPoints = this.map.get(forkId) ?? new Set();
+        exitPoints.add(blockHeight);
+        this.map.set(forkId, exitPoints);
+
+        return this.map.get(forkId) as Set<BlockHeight>;
+    }
+
+    // ====================================
+    // READ
+    // ====================================
+
+    getExitPointsInRange(
+        forkId: ForkId,
+        start?: BlockHeight,
+        end?: BlockHeight
+    ): BlockHeight[] {
+        const exitPoints = this.map.get(forkId);
+        if (!exitPoints?.size) {
+            return [];
+        }
+
+        // invalid range
+        if (start !== undefined && end !== undefined && end <= start) {
+            return [];
+        }
+        const list = Array.from(exitPoints).sort(
+            (a, b) => Number(a) - Number(b)
+        );
+
+        if (start === undefined && end === undefined) {
+            return list;
+        }
+
+        const actualStart = start ?? list[0];
+        const actualEnd = end ?? list[list.length - 1] + 1;
+
+        const filteredExitPoints = list.filter(
+            (height) => height >= actualStart && height < actualEnd
```

Comment:
To consider maybe best to do `[low,high]` instead of `[low,high)` even though most conventions do [low,high) since we'll be calling this mostly in the form `[firstBlock.height, latestBlock.height]` where if the lastBlock also produced an exit we want to account for it

Either way even if we have to do a +1 it's fine

Reply:
done